src/BloomExe/ImageProcessing/RuntimeImageProcessor.cs

```diff
@@ -90,114 +89,148 @@ public string GetPathToResizedImage(string originalPath)
 				}
 
-				using (var originalImage = PalasoImage.FromFile(originalPath))
+				// there is not a cached version, try to make one
+				var pathToProcessedImage = Path.Combine(_cacheFolder, Path.GetRandomFileName() + Path.GetExtension(originalPath));
+
+				if (!Directory.Exists(Path.GetDirectoryName(pathToProcessedImage)))
+					Directory.CreateDirectory(Path.GetDirectoryName(pathToProcessedImage));
+
+				// BL-1112: images not loading in page thumbnails
+				var success = false;
+				if (getThumbnail)
 				{
-					//if it's a jpeg, we don't resize, we don't mess with transparency, nothing. These things
-					//are scary in .net. Just send the original back and wash our hands of it.
-					if (ImageUtils.AppearsToBeJpeg(originalImage))
-					{
-						return originalPath;
-					}
+					// The HTML div that contains the thumbnails is 80 pixels wide, so make the thumbnails 80 pixels wide
+					success = GenerateThumbnail(originalPath, pathToProcessedImage, 80);
+				}
+				else
+				{
+					success = MakeTransparentImage(originalPath, pathToProcessedImage);
+				}
+
+				if (!success) return originalPath;
+
+				try
+				{
+					_originalPathToProcessedVersionPath.Add(cacheFileName, pathToProcessedImage); //remember it so we can reuse if they show it again, and later delete
+				}
+				catch (ArgumentException)
+				{
+					// it happens sometimes that though it wasn't in the _originalPathToProcessedVersionPath when we entered, it is now
+					// I haven't tracked it down... possibly we get a new request for the image while we're busy compressing it?
+				}
 
-					double shrinkFactor = 1.0;
+				return pathToProcessedImage;
+			}
+		}
 
-#if ShrinkLargeImages // at the moment, we're suspecting that it may be better to let the browser do the shrinking
+		private bool GenerateThumbnail(string originalPath, string pathToProcessedImage, int newWidth)
+		{
+			try
+			{
+				using (var originalImage = PalasoImage.FromFile(originalPath))
+				{
+					// calculate dimensions
+					var newW = (originalImage.Image.Width > newWidth) ? newWidth : originalImage.Image.Width;
+					var newH = newW * originalImage.Image.Height / originalImage.Image.Width;
 
-					//if its a small image, like a creative commons logo, we don't try and resize it
-					if (originalImage.Image.Width > TargetDimension || originalImage.Image.Height > TargetDimension)
+					using (var newImg = originalImage.Image.GetThumbnailImage(newW, newH, () => false, IntPtr.Zero))
 					{
-						var maxDimension = Math.Max(originalImage.Image.Width, originalImage.Image.Height);
-						//enhance: if we had a way of knowing what the target dimension actually was, we'd use that, of course
-						shrinkFactor = (TargetDimension/(double) maxDimension);
+						newImg.Save(pathToProcessedImage);
 					}
-#endif
-					var destWidth = (int) (shrinkFactor*originalImage.Image.Width);
-					var destHeight = (int) (shrinkFactor*originalImage.Image.Height);
+				}
+			}
+			catch (OutOfMemoryException)
```

Comment:
This kind of scares me. Does it really help? If we're out of memory while generating a thumbnail, seems pretty likely we'll soon be out of it anyway, especially since the caller will now attempt to use the original, larger image.